### PR TITLE
add failing tests for pack with files in directory

### PR DESCRIPTION
**Pull request by t-obi**
**Summary**

pack seems to be broken if the package contains files in directories, see #1442 .
This PR just adds failing tests to illustrate the problem.

**Test plan**

Run yarn tests to see the failing tests

### CODE REVIEW

**Comment by t-obi** [comment]
It's also weird that different tests are failing and passing on travis compared to locally on my system.
On circleci the same tests are passing and failing as locally for me.

File: yarn.lock [hunk 1]
```diff
@@ -4705,6 +4712,14 @@ tapable@^0.2.3, tapable@~0.2.3:
   version "0.2.4"
   resolved "https://registry.yarnpkg.com/tapable/-/tapable-0.2.4.tgz#a7814605089d4ba896c33c7e3566e13dcd194aa5"
 
+tar-fs:
```
Comment (bestander) [inline yarn.lock]: Could you use latest master branch to run `yarn add tar-fs` for this change?
In 0.16 we had an issue that generated a lockfile entry without version

**Comment by t-obi** [comment]
Thanks for having a look at this!
I have rebased and ran yarn add tar-fs with current master.
The lint-error should also be gone now.

This is still work in progress though and should probably not be merged yet because some things are not implemented, for example bundled dependencies and the "files" field in package.json should not work at all.
It would be great to get some feedback on this though, because I'm not sure if i'm going in the right direction at all. For example I was not sure if it is ok to add tar-fs as a dependency, but it made everything simpler in my opinion.

File: __tests__/commands/pack.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
+/* @flow */
```
Comment (bestander) [inline __tests__/commands/pack.js]: Thanks for tests 👍

File: package.json [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "semver": "^5.1.0",
     "strip-bom": "^2.0.0",
     "tar": "^2.2.1",
+    "tar-fs": "^1.14.0",
     "tar-stream": "^1.5.2",
```
Comment (bestander) [inline package.json]: I think tar-stream is not used anymore

**Comment by bestander** [comment]
I'll close this for now because no one got back to this for a long time.
Feel free to reopen

**Pull request closed without merging**